pkg/providers/amazon/autoscaling/autoscaling.go

```diff
@@ -99,10 +99,12 @@ func (group *Group) getSpotRequests() ([]*ec2.SpotInstanceRequest, error) {
 	}
 
 	lc, err := group.getLaunchConfiguration()
-	if err == nil && lc == nil {
-		err = errors.NewWithDetails("could not find launch configuration for ASG", "asg", group.getName())
-	}
 	if err != nil {
+		if _, ok := err.(errLaunchConfigurationNotFound); ok {
+			if group.LaunchTemplate != nil {
+				return nil, nil
```

Comment:
**Question**: in this case, couldn't (shouldn't?) we check the launch template for the same things we check the launch configuration for during the logic following this point in the function (spot price, spot instance requests)?

Are they so different we can't do anything here?
Because if we could, I would sleep better knowing we do check it otherwise in EKS cases because of the launch template we are going to skip those checks entirely and silently with this shortcut.

I have a feeling spot requests should be query-able from launch templates as well (maybe through some other resources created from launch templates if not directly).

To clarify, I just don't like returning `nil, nil` from a `getSpotRequests()` function when there is a launch template with potential spot requests to check (first because it is less idiomatic, but this is the least of my problems, second because of the not checked/verified spot requests).
Maybe if we could properly retrieve the launch template spot requests we wouldn't need to be manually checking for spot errors in the `WaitForASGToBeFulfilled()` function, because we would already got our error from the spot request checking in the ASG somewhere around here.

Reply:
I checked and I only found spot requests related to PKE clusters, so I thought things work differently when launch templates are being used, but I couldn't yet figure out exactly how.

There is another `nil, nil` return a bit downwards, that is way I went with this. In the function that calls this one, there is a check for this scenario, but it would be nice to handle things a different way for sure.

Reply:
Okay, well done then, I'm going to double check just to make sure (stand by) but if we can't do it any better way, this will have to suffice.

Reply:
The only thing I found is that the spot price is part of the launch template **version's** description, it could be queried from there in case of an LT. This can be a substitute in the `getSpotRequests()` function in place of the launch configuration's spot price, but as it is already part of the ASG last activity error, there is no need for further development.

The spot requests can be queried from the EC2 API's DescribeSpotInstanceRequests, but it seems like even without filtering I could not find the LT's spot requests, it is possible the ASG requests instances directly so we cannot build pretty requests like in the case of PKE.

Because of this I'm all for letting the spot requests go and use the ASG activity for informational purposes.